websocket.go

```diff
@@ -531,7 +533,9 @@ func (c *wsConn) handleWsConn(ctx context.Context) {
 	// setup pings
 
 	stopPings := c.setupPings()
-	defer stopPings()
+	defer func() {
+		stopPings()
```

Comment:
So, this code _was_ incorrect. However, it's now racy. Before, `stopPings` was:

1. Read exactly once here.
2. Written from within a goroutine, under the write lock.

Now, it's read when this function returns, when the write lock isn't held.

Reply:
When the `stopPings`  is re-assigned in `c.incoming` case , `stop` channel  cannot be closed properly

Reply:
I know, I'm just saying that this patch isn't the right fix.